spec/layers.md

```diff
@@ -0,0 +1,99 @@
+clear function
+==============
+**Status**: explainer.
+
+Provide a `beginLayer()` and `endLayer()` functions that create and close layers for Canvas.
+
+
+Rationale
+---------
+
+Currently, the only way to draw a series of drawing calls at the same time, while applying the same sort of effects (shadows, blur, alpha, filters, compositing) to all of the drawn objects, without having these effects being applied one to one to this object, is by using an auxiliary canvas and then, drawing that auxiliary canvas to the canvas.
+This proposal adds an alternative to that (the same way that Fullter in Google and Core Graphics at Apple do) by creating a layer that then will be drawn in one go when the layer ends.
+
+
+Proposal
+--------
+
+```webidl
+interface CanvasRenderingContext2D {
+  // extending original
+  void beginLayer();
+  void endLayer();
+};
+
+interface OffscreenCanvasRenderingContext2D {
+  // extending original
+  void beginLayer();
+  void endLayer();
+};
+```
+The rendering of the layer will use the current state of the canvas, and will reset these effects to the defaults at the end of the layer. The effects that will be used are:
+- globalAlpha
+- globalCompositeOperation
+- shadowOffsetX
+- shadowOffsetY
+- shadowColor
+- shadowBlur
+- filter
+
+`beginLayer()` sets the start point of the layer, it also captures the current state of the canvas (see list above) that will be used when rendering the layer.
+`endLayer()` sets the end point of the layer. At that moment the layer itself will be drawn as one single object into the canvas. When the layer ends, we reset all the effects used in the drawing of the layer at the end of it (see list above).
+
+These methods are nesteable, so layers can be created and drawn within layers.
+
+If there is a dangling `beginLayer()` without `endLayer()`, those operations will not be drawn.
+
+For the interactions with `save()` and `restore()`, a `save()` without `restore()` (or single `restore()` without `save()`) between a pair of `beginLayer()` and `endLayer()` will be discarded.
```

Comment:
Is this necessary? It feels like that will mix badly with `clip()` (which still can't be discarded without save/restore).

Wouldn't it be easier to state that layers are not saved by `save()`, and that they keep doing what they are doing?